File: 268. Missing Number/268. Missing Number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
  const set = new Set();

  for (num of nums) {
    set.add(num);
  }

  for (let i = 0; i < nums.length + 1; i++) {
    if (!set.has(i)) return i;
  }
};

File: 268. Missing Number/268. Missing Number.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int expected = (n * (n + 1)) / 2;
        int actual = 0;
        for(int num : nums){
            actual += num;
        }     
        return expected - actual;
    }
}